### ISSUE TITLE
CLJS server side to reduce incidental complexity

### ISSUE THREAD

**Issue by learningcljs** [magomimmo/modern-cljs#339]
In [tutorial 12](https://github.com/magomimmo/modern-cljs/blob/master/doc/second-edition/tutorial-12.md) it says:

> No way that anybody could convince me that there is something better than CLJ/CLJS on Planet Web.

This statement raises this question for me:  What about CLJS on both client and server side?

[Tutorial 13](https://github.com/magomimmo/modern-cljs/blob/master/doc/second-edition/tutorial-13.md) mentions the incidental complexity of having to deal with CLJ on the sever side and CLJS on the client side.

> But we're in a good position to cover the missing step, because we have **the same language on both sides**. It should not be a PITA to go from one side to the other and vice-versa. **Just be prepared to pay a little attention anytime you cross the border in either direction**.

The [differences between CLJ and CLJS](https://github.com/clojure/clojurescript/wiki/Differences-from-Clojure) make them seem like they are not quite the same language.  Which is another source of incidental complexity.

In [tutorial 14](https://github.com/magomimmo/modern-cljs/blob/master/doc/second-edition/tutorial-14.md) this incidental complexity is again demonstrated:

> We know from the above tutorial that we can share the validation rules between the server and the client sides by creating a portable CLJ/CLJS source file with the **.cljc extension** in the src/cljc source directory's structure.

> To write a portable testing namespace for the portable modern-cljs.shopping.validators namespace we created above, we have to use the same **#? trick to differentiate the different testing namespace declarations used by CLJ and CLJS**.

I can appreciate that this incidental complexity is less than previous methods.

But can this incidental complexity be reduced or eliminated by using CLJS on both the client and server side as is hinted to in the [README](https://github.com/magomimmo/modern-cljs/blob/master/README.md):

> CLJS is a compiler for the Clojure programming language that targets JavaScript. It emits JavaScript code which runs in web browsers and other client-side or server-side JavaScript interpreters (e.g. nodejs).

I think a couple of benefits of being able to reduce or eliminate this incidental complexity could be decreased learning curve and a simpler code base.

I am sure like everything that there would be trade offs with going for CLJS on both the client and server.  I would be curious what some of the disadvantages of using CLJS on the server side would be.

**Comment by learningcljs** [comment]
In regards to magomimmo/modern-cljs#338:

> It's not impossible that in the future I'll create a new series for nodejs, because I'm working hard on a cljs project which uses nodejs

I found myself thinking as I was working through this tutorial series that it would be very nice to have a tutorial series that covers CLJS on both client and server side.  It could be a 2b or 3 edition of this tutorial series or a new one.

**Comment by magomimmo** [comment]
I agree, it would be very nice. At the same time at the moment  I do not have the time and the energy to expand the series in this direction. My team is working on a very big project which uses nodejs/cljs on a micro-controller (full of core.async stuff). It uses clj/hadoop suite on the backend and finally react/react-native (used from reagent) for the UI (destop, web, ios and android). I plan to write something about this very enriching experience not before 9/12 months. Sorry about that.
